test/plugin/infer_extension_test.rb

```diff
@@ -59,12 +59,28 @@
     uploaded_file = @uploader.upload(fakeio(content_type: "image/jpeg"))
     assert_equal ".jpeg", File.extname(uploaded_file.id)
     assert_nil uploaded_file.original_filename
+  end
 
+  it "does not replace existing extension when generating location" do
     uploaded_file = @uploader.upload(fakeio(filename: "nature.jpg", content_type: "image/jpeg"))
     assert_equal ".jpg", File.extname(uploaded_file.id)
     assert_equal "nature.jpg", uploaded_file.original_filename
   end
 
+  describe "force: true" do
+    before do
+      @uploader = uploader { plugin :infer_extension, force: true }
+    end
+
+    it "replaces with inferred extension when generating location" do
+      uploaded_file = @uploader.upload(fakeio(filename: "nature.wrong", content_type: "image/jpeg"))
+      assert_equal ".jpeg", File.extname(uploaded_file.id)
+
+      uploaded_file = @uploader.upload(fakeio(filename: "nature.mp3", content_type: "image/jpeg"))
+      assert_equal ".jpeg", File.extname(uploaded_file.id)
```

Comment:
It would be nice to have a test showing canonicalizing of extensions. 
What happens if the filename has no extension - nothing breaks right?